File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function(nums) {
    let a =[];
    for(let i=0;i<nums.length;i++){
        a[i] = nums[i]**2;
    }
    a.sort((a,b)=> a-b);
    return a
};

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution(object):
    def sortedSquares(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        i =0
        j =len(nums)-1
        arr = [0]*len(nums)
        k = len(nums)-1
        while i<=j:
            sum1 = nums[i]**2
            sum2 = nums[j]**2
            if sum1<sum2:
                arr[k] = sum2
                j-=1
            else:
                arr[k] = sum1
                i+=1
            k-=1
        return arr


          